app/src/Pages/DetailsPage/Details.tsx

```diff
@@ -65,7 +71,10 @@ function Details() {
   const handleNewPriceHistorySubmission = (newPriceHistorySubmission: PriceHistory) => {
     console.log("price history state before adding stuff in: ", priceHistory); 
     setPriceHistory(priceHistory => [newPriceHistorySubmission, ...priceHistory])
-    console.log("price history state after adding stuff in: ", priceHistory); 
+    console.log("price history state after adding stuff in: ", priceHistory);
+    setSalePrice(newPriceHistorySubmission.price); 
+    let avgPrice: number = findAverageCardPrice([newPriceHistorySubmission, ...priceHistory]);
```

Comment:
Do you know why we need to pass in the appended version of the price history, when we had already appended it a few lines above? I realized it is required to pass in this appended version for it to work, but was just wondering why

Reply:
Also, if we do go the route of having a string message appear when there is no price entries, I think that we should add a check right here to see if the length of the price history array is 0. That way, we don't divide by 0 in average-calculation

Reply:
> you know why we need to pass in the appended version of the price history, when we had already appended it a few lines above? I realized it is required to pass in this

I am really not sure why, but this was what was working I think it has to do with rendering.